sklearnex/linear_model/incremental_linear.py

```diff
@@ -0,0 +1,177 @@
+# ===============================================================================
+# Copyright 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===============================================================================
+from daal4py.sklearn._utils import daal_check_version
+
+if daal_check_version((2023, "P", 100)):
+    import numpy as np
+    from sklearn.utils import check_array, gen_batches
+
+    from daal4py.sklearn._utils import control_n_jobs, run_with_n_jobs
+    from onedal._device_offload import support_usm_ndarray
+    from onedal.linear_model import (
+        IncrementalLinearRegression as onedal_IncrementalLinearRegression,
+    )
+
+    @control_n_jobs
+    class IncrementalLinearRegression:
+        """
+        Incremental estimator for linear regression.
+        Allows to train linear regression if data are splitted into batches.
+        Parameters
+        ----------
+        fit_intercept : bool, default=True
+        Whether to calculate the intercept for this model. If set
+        to False, no intercept will be used in calculations
+        (i.e. data is expected to be centered).
+
+        copy_X : bool, default=True
+            If True, X will be copied; else, it may be overwritten.
+
+        n_jobs : int, default=None
+            The number of jobs to use for the computation.
+
+        batch_size : int, default=None
+            The number of samples to use for each batch. Only used when calling
+            ``fit``. If ``batch_size`` is ``None``, then ``batch_size``
+            is inferred from the data and set to ``5 * n_features``, to provide a
+            balance between approximation accuracy and memory consumption.
+
+        Attributes
+        ----------
+        coef_ : array of shape (n_features, ) or (n_targets, n_features)
+            Estimated coefficients for the linear regression problem.
+            If multiple targets are passed during the fit (y 2D), this
+            is a 2D array of shape (n_targets, n_features), while if only
+            one target is passed, this is a 1D array of length n_features.
+
+        intercept_ : float or array of shape (n_targets,)
+            Independent term in the linear model. Set to 0.0 if
+            `fit_intercept = False`.
+
+        n_features_in_ : int
+            Number of features seen during :term:`fit`.
+
+        """
+
+        def __init__(self, fit_intercept=True, copy_X=True, n_jobs=None, batch_size=None):
```

Comment:
What is the purpose of `batch_size`? It can be deducted from input data.

Reply:
it is described in docstring few lines above. this is copy from sklearn.IncrementalPCA interface